jetty-ee10/jetty-ee10-webapp/src/main/resources/org/eclipse/jetty/ee10/webapp/catalog-ee10.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog" xml:base="">
+  <!-- 2.2 -->
+  <public publicId="web-app_2_2.dtd" uri="resources/web-app_2_2.dtd" />
```

Comment:
Shouldn't this be:
```suggestion
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog" xml:base="jakarta/servlet">
  <!-- 2.2 -->
  <public publicId="web-app_2_2.dtd" uri="resources/web-app_2_2.dtd" />
```

or

```suggestion
<catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog" xml:base="">
  <!-- 2.2 -->
  <public publicId="web-app_2_2.dtd" uri="jakarta/servlet/resources/web-app_2_2.dtd" />
```

Reply:
Aaah!! The URI is resolved relative to a specific class (eg Servlet.class).   Hmmm OK, but if we were to ask the servlet spec to include the catalog, it would probably be relative to the top of the jar?

Resolving against a class is good... but is there a way the contents of this file can be relative to the top level?

Reply:
No, the `catalog.xml` that the `servlet-api-<ver>.jar` would live in the `jakarta/servlet/catalog.xml` location (for Servlet 5+)

That would make the URI to the catalog something like ...

```
jar:file://path/to/jakarta.servlet.api-6.0.1.jar!/jakarta/servlet/catalog.xml
```

Which would also be the baseURI for the catalog.
All non-absolute URI references are relative to that baseURI.

Basically doing a normal Base URI resolve techniques.

```
base  : jar:file://path/to/jakarta.servlet.api-6.0.1.jar!/jakarta/servlet/catalog.xml
ref   : resources/web-app_6_0.xsd
result: jar:file://path/to/jakarta.servlet.api-6.0.1.jar!/jakarta/servlet/resources/web-app_6_0.xsd
```

Reply:
ack